field_audit/models.py

```diff
@@ -636,15 +636,24 @@ def delete(self, *, audit_action=AuditAction.RAISE):
         from .field_audit import request
         request = request.get()
         audit_events = []
-        for instance in self:
-            # make_audit_event_from_instance cannot return None when delete=True
-            audit_events.append(AuditEvent.make_audit_event_from_instance(
-                instance,
-                False,
-                True,
-                request,
-                instance.pk,
-            ))
+        fields_to_fetch = set(
+            getattr(self.model, AuditEvent.ATTACH_FIELD_NAMES_AT)
+        )
```

Comment:
I think I made the `AuditEvent._field_names()` method private because I was still unsure if it would remain the "final" way to do that. At this point you've hammered out more of the design that solidifies the way field names are being stored on the model class, so I think it would be good to "unprivatize" that method and use it here.

(Extra credit, the `update()` method could then also use it).

```suggestion
        fields_to_fetch = set(AuditEvent.field_names(self.model))
```